src/problem_depot/problems/sdp.jl

```diff
@@ -1361,3 +1361,35 @@ end
         end
     end
 end
+
+@add_problem sdp function sdp_min_maxeig_canon_lmi(handle_problem!, ::Val{test}, atol, rtol, ::Type{T}) where {T, test}
+    # Minimize the maximum eigenvalue of an affine matrix function of a vector argument. Formulated
+    # as a linear optimization subject to a canonical LMI (SDP) constraint:
+    #
+    # minimize      λ
+    # subject to    A(x)⪯λ
+    # where x is a vector in Rⁿ and the matrix function A(x) = A₀ + A₁x₁ + A₂x₂ + … + Aₙxₙ.
+    #
+    # Besides serving as a benchmark, it also tests the solution of the Issue
+    # https://github.com/jump-dev/Convex.jl/issues/447
+    # (the issues boils down to  evaluation of `A₁x₁`, because `x₁` was a matrix of size (1,1)).
+
+    A₀ = [1.0 2.0; 2.0 3.0]
+    A₁ = [1.0 0.0; 0.0 -1.0]
+    A₂ = [4.0 5.0; 5.0 -6.0]
+
+    x = Variable(2)
+    λ = Variable()
+
+    A = A₀ + A₁*x[1] + A₂*x[2]
+
+    p = minimize(λ,A⪯λ; numeric_type=T)
```

Comment:
I think the problem here actually does something different: it minimizes λ such that `λ .- A` is positive semi-definite (note the `.-` instead of `-`). I hope this has not affected your or other's results; I see how it's very confusing.
```suggestion
    p = minimize(λ,A⪯λ*eye(2); numeric_type=T)
```
Maybe we should add this warning somewhere to the docs or examples? And/or deprecate the current behavior and error when there's both matrices and scalars in PSD constraints.

Here I'm using `eye(2)` which is defined in the `problem_depot.jl` file. Usually I'd use `using LinearAlgebra; I(2)` but that doesn't work on Julia 1.0.

Reply:
In fact, I originally quickly tried (after `using LinearAlgebra`)

```julia
julia> problem = minimize(λ,A⪯λ*I)
ERROR: MethodError: no method matching *(::Variable, ::UniformScaling{Bool})
Closest candidates are:
  *(::Any, ::Any, ::Any, ::Any...) at operators.jl:560
  *(::Union{MathOptInterface.ScalarAffineFunction{T}, MathOptInterface.ScalarQuadraticFunction{T}, MathOptInterface.VectorAffineFunction{T}, MathOptInterface.VectorQuadraticFunction{T}}, ::T) where T at /home/hurak/.julia/packages/MathOptInterface/YDdD3/src/Utilities/functions.jl:2995
  *(::Convex.AbstractExpr, ::Union{Number, AbstractArray}) at /home/hurak/.julia/dev/Convex/src/atoms/affine/multiply_divide.jl:103
  ...
Stacktrace:
 [1] top-level scope
   @ none:1
```
and seeing it fail (I think I still do not quite master the `UniformScaling` operator), rather than going with the ugly `Matrix(1.0I, 2, 2) ` (I was not even aware of the possibility of the usage `I(2)`), I just tried going with just a scalar on the r.h.s. And I did not realize that broadcasting will be applied here. 

And no, it did not affect my other results since the whole example is artificially crafted only to demonstrate the bug. That's why I did not discover that it does not compute what it should. I only blindly focused on the bug and did not think about the actual value of the result.

Reply:
okay, glad to hear it wasn't a problem! Would be nice also if we could support UniformScaling's directly, maybe I'll file an issue for that.

Reply:
Only now I realized that with the changed problem formulation, the correct solutions that I hardwired into the tests must be changed. Honestly, I do not know how to proceed formally now, after the PR has been submitted (I am still not quite fluent with this git/github collaborative development) and I do not want to screw it up. I will perhaps only put the correct lines here:
```julia
if test
    @test λ.value ≈ 2.400000051025101 atol=atol rtol=rtol
    @test x.value ≈ [3.0000000535867315, -0.4000000018585541] atol=atol rtol=rtol
    @test evaluate(A) ≈ [2.400000046152515 -9.292770553059881e-9; -9.292770553059881e-9 2.399999957564593] atol=atol rtol=rtol
end
```
They should go to the end of src/problem_depot/problems/sdp.jl.

Reply:
Sure, so all you need to do is make another commit on your branch with those changes, and do `git push`. It will automatically update the PR here. (In general GitHub will keep the PR in-sync with your branch, so any changes you do there should be reflected here). Also, just so you know, clicking "resolve conversation" doesn't actually make any changes, it just hides the comments. (Clicking "commit suggestion" does make changes though).